pages/index.tsx

```diff
@@ -0,0 +1,245 @@
+'use client';
+import { useEffect, useState } from 'react'
+import axios from "axios";
+import Prefectures from "@/utils/Pref";
+import { Select } from 'antd';
+import { MdOutlineLocationOn, MdWbSunny } from "react-icons/md";
+import { MdMyLocation } from "react-icons/md";
+import { WiHumidity,WiStrongWind } from "react-icons/wi";
+import  ForecastWeatherDetail  from "@/components/ForecastWeatherDetails";
+import { format, parseISO } from "date-fns";
+import { LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip,Legend } from 'recharts';
+import { useSearchParams, usePathname, useRouter } from 'next/navigation';
+
+// Format in which current weather api returns data for given location
+interface WeatherDetail {
+  dt: number;
+  dt_txt:string;
+  main: {
+    temp: number;
+    feels_like: number;
+    temp_min: number;
+    temp_max: number;
+    humidity: number;
+  };
+  weather: {
+    id: number;
+    main: string;
+    description: string;
+    icon: string;
+  }[];
+  clouds: {
+    all: number;
+  };
+  wind: {
+    speed: number;
+    deg: number;
+    gust: number;
+  };
+}
+
+// format in which forecast for 5 days api returns the data for a given location
+interface WeatherData {
+  cod: string;
+  message: number;
+  cnt: number;
+  list: WeatherDetail[];
+  city: {
+    id: number;
+    name: string;
+    coord: {
+      lat: number;
+      lon: number;
+    };
+    country: string;
+    population: number;
+    timezone: number;
+    sunrise: number;
+    sunset: number;
+  };
+}
+
+// props of Linechart
+interface Chart {
+  dates:string,
+  temp:number
+}
+
```

Comment:
These types should be in a separate file! You can look at how we structure things in TimeTrackPro repo.

Reply:
Yes saw the time track pro code, moved all these to a type folder. Will follow this structure from next time ! Made the changes now.